test/org/sagebionetworks/bridge/services/AccountWorkflowServiceTest.java

```diff
@@ -217,6 +256,57 @@ public void requestResetPassword() throws Exception {
     }
     
     @Test
+    public void requestResetPasswordWithPhone() throws Exception {
+        when(service.createTimeLimitedToken()).thenReturn("ABC");
+        when(mockAccountDao.getAccount(ACCOUNT_ID_WITH_PHONE)).thenReturn(mockAccount);
+        when(mockStudyService.getStudy(TEST_STUDY)).thenReturn(study);
+        when(mockAccount.getPhone()).thenReturn(TestConstants.PHONE);
+        when(mockAccount.getPhoneVerified()).thenReturn(Boolean.TRUE);        
+        when(mockAccount.getStudyIdentifier()).thenReturn(TEST_STUDY);
+        
+        service.requestResetPassword(study, ACCOUNT_ID_WITH_PHONE);
+        
+        verify(mockCacheProvider).setString(eq("ABC:phone:api"), stringCaptor.capture(), eq(60*60*2));
+        
+        Phone captured = BridgeObjectMapper.get().readValue(stringCaptor.getValue(), Phone.class);
```

Comment:
stringCaptor is used twice here, so we never validate what's actually being written to mockCacheProvider.

Reply:
Re-arranged these